### PR TITLE
Make it possible to pass data to templates

### PR DESCRIPTION
**Pull request by kjarmicki**
This PR adds ability to pass additional data to templates that cannot or should not be obtained with questions.

My particular case is: I have a template inside of a module and I'd like to get that module's version from `package.json`.

### CODE REVIEW

**Comment by kjarmicki** [comment]
Hm.. interesting, build failed on a file that I haven't touched. Not reproducable locally yet, but I'll keep trying :thinking:

File: docs/Templates.md [hunk 1]
```diff
@@ -161,6 +161,19 @@ The `prompts` property follows [Inquirer.js](https://github.com/SBoudrias/Inquir
 
 `name` and `author` are a bit special and will automatically default to the directory name and, if available, the name from the git config.
 
+#### `data`
```
Comment (dlmr) [inline docs/Templates.md]: This section should be below `##### 'when'`.

File: docs/Templates.md [hunk 2]
```diff
@@ -161,6 +161,19 @@ The `prompts` property follows [Inquirer.js](https://github.com/SBoudrias/Inquir
...
+`data` property contains any arbitrary data you'd like to pass to templates. It supports both simple values and async functions / promises that eventually get resolved to values.
+Note that functions are supported only when using `.js` version of a config.
+
+```javascript
+{
+  "data": {
+    "version": "1.0.2",
+    "recentChanges": fetch("https://some-changelog.org/recent")
```
Comment (dlmr) [inline docs/Templates.md]: Given how the code is structured this example should be:
```js
"recentChanges":  () => fetch("https://some-changelog.org/recent")
```

File: src/commands/init/generateTemplate/collect.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    }
+
+    done();
+}
+
+async function unwrap(metadata, key, dataProvider) {
+    /* eslint-disable no-param-reassign */
+    if (typeof dataProvider === 'function') {
+        try {
+            metadata[key] = await dataProvider();
```
Comment (dlmr) [inline src/commands/init/generateTemplate/collect.js]: What do you think about passing in the `metadata` here, into `dataProvider()`?

Then we could get access to the answers that the user has answered.

```js
// Provide metadata
await dataProvider(metadata);

// Later we can use it
"recentChanges":  (answers) => fetch("https://some-changelog.org/recent")
```

If we do this change we should also add a comment about it in the documentation.

**Review by dlmr — CHANGES_REQUESTED** [review]
Thanks for the PR and sorry for being slow on getting back to this.

I like the change, just some small comments on ways we can make it a little better.

File: src/commands/init/generateTemplate/collect.js (hunk 3 above)
Comment (kjarmicki) [reply src/commands/init/generateTemplate/collect.js]: Good point :)

**Review by dlmr — APPROVED** [review]

**Pull request merged**